src/dnx.coreclr/dnx.coreclr.cpp

```diff
@@ -397,126 +232,128 @@ extern "C" HRESULT __stdcall CallApplicationMain(PCALL_APPLICATION_MAIN_DATA dat
+        if (!GetTrustedPlatformAssembliesList(core_clr_directory, false, trusted_platform_assemblies))
         {
-            printf_s("Failed to find files in the coreclr directory\n");
+            trace_writer.write(L"Failed to find TPA files in the coreclr directory", false);
             return E_FAIL;
         }
     }
 
-    // Add the assembly containing the app domain manager to the trusted list
-
-    errno = wcscat_s(pwszTrustedPlatformAssemblies, cchTrustedPlatformAssemblies, szCurrentDirectory);
-    CHECK_RETURN_VALUE_FAIL_EXIT_VIA_FINISHED(errno);
-
-    errno = wcscat_s(pwszTrustedPlatformAssemblies, cchTrustedPlatformAssemblies, L"dnx.coreclr.managed.dll");
-    CHECK_RETURN_VALUE_FAIL_EXIT_VIA_FINISHED(errno);
+    auto current_directory = data->runtimeDirectory ? data->runtimeDirectory : GetModuleDirectory(NULL);
 
-    //wstring appPaths(szCurrentDirectory);
-    WCHAR wszAppPaths[MAX_PATH];
-    wszAppPaths[0] = L'\0';
-
-    errno = wcscat_s(wszAppPaths, _countof(wszAppPaths), szCurrentDirectory);
-    CHECK_RETURN_VALUE_FAIL_EXIT_VIA_FINISHED(errno);
-
-    errno = wcscat_s(wszAppPaths, _countof(wszAppPaths), L";");
-    CHECK_RETURN_VALUE_FAIL_EXIT_VIA_FINISHED(errno);
-
-    errno = wcscat_s(wszAppPaths, _countof(wszAppPaths), szCoreClrDirectory);
-    CHECK_RETURN_VALUE_FAIL_EXIT_VIA_FINISHED(errno);
+    // Add the assembly containing the app domain manager to the trusted list
+    trusted_platform_assemblies.append(dnx::utils::path_combine(current_directory, L"dnx.coreclr.managed.dll"));
 
-    errno = wcscat_s(wszAppPaths, _countof(wszAppPaths), L";");
-    CHECK_RETURN_VALUE_FAIL_EXIT_VIA_FINISHED(errno);
+    std::wstring app_paths;
+    app_paths.append(current_directory).append(L";");
+    app_paths.append(core_clr_directory).append(L";");
 
     const wchar_t* property_values[] = {
         // APPBASE
         data->applicationBase,
         // TRUSTED_PLATFORM_ASSEMBLIES
-        pwszTrustedPlatformAssemblies,
+        trusted_platform_assemblies.c_str(),
         // APP_PATHS
-        wszAppPaths,
+        app_paths.c_str(),
         // Use the latest behavior when TFM not specified
         L"UseLatestBehaviorWhenTFMNotSpecified",
     };
 
     DWORD domainId;
-    DWORD dwFlagsAppDomain =
-        APPDOMAIN_ENABLE_PLATFORM_SPECIFIC_APPS |
-        APPDOMAIN_ENABLE_PINVOKE_AND_CLASSIC_COMINTEROP |
-        APPDOMAIN_DISABLE_TRANSPARENCY_ENFORCEMENT;
 
-    LPCWSTR szAssemblyName = L"dnx.coreclr.managed, Version=0.1.0.0";
-    LPCWSTR szEntryPointTypeName = L"DomainManager";
-    LPCWSTR szMainMethodName = L"Execute";
-
-    int nprops = sizeof(property_keys) / sizeof(wchar_t*);
-
-    hr = pCLRRuntimeHost->CreateAppDomainWithManager(
+    HRESULT hr = pCLRRuntimeHost->CreateAppDomainWithManager(
         L"dnx.coreclr.managed",
-        dwFlagsAppDomain,
+        APPDOMAIN_ENABLE_PLATFORM_SPECIFIC_APPS | APPDOMAIN_ENABLE_PINVOKE_AND_CLASSIC_COMINTEROP | APPDOMAIN_DISABLE_TRANSPARENCY_ENFORCEMENT,
```

Comment:
I can do that but it does not really matter - it won't make some of the flags less cryptic (besides, this way it almost looks like `BindingFlags` :smiley:)